src/cpp/src/sequence_group.hpp

```diff
@@ -206,6 +206,8 @@ class SequenceGroup {
     size_t m_max_content_len = 0;
     // max validation length within a group to check generated tokens
     size_t m_num_validated_tokens = 0;
+    // flag to enable/disable token generation, e.g. in speculative decoding scenario
+    bool m_is_gen_paused = false;
```

Comment:
we already have `m_status` for sequence group, where we can set `WAITING`. Can we re-use it here?

Reply:
for Sequence, not for for SequenceGroup. It was the main reason is to implement such approach

Reply:
Discussed offline
Agreed to save the option